src/ansys/mapdl/core/launcher.py

```diff
@@ -1353,12 +1360,31 @@ def launch_mapdl(
         raise ValueError(f"The following arguments are not recognaised: {ms_}")
 
     if ip is None:
-        ip = os.environ.get("PYMAPDL_IP", LOCALHOST)
+        ip = os.environ.get("PYMAPDL_IP", None)
+
+        if not ip and ON_WSL:
+            ip = _get_windows_host_ip()
+            if ip:
+                LOG.debug(
+                    f"On WSL: Using the following IP for the Windows OS host: {ip}"
+                )
+            else:
+                LOG.debug("PyMAPDL could not find the IP of the Windows Host Machine.")
+
+        if not ip:
+            LOG.debug(f"No IP supplied, using default IP: {LOCALHOST}")
+            ip = LOCALHOST
+
     else:
         LOG.debug(
-            "Because ``PYMAPDL_IP is not None, an attempt is made to connect to a remote session. ('START_INSTANCE' is set to False.`)"
+            "Because ``PYMAPDL_IP is not None, an attempt is made to connect to"
+            " a remote session. ('START_INSTANCE' is set to False.`)"
```

Comment:
```suggestion
            "Because 'PYMAPDL_IP' is not None, an attempt is made to connect to"
            " a remote session ('START_INSTANCE' is set to 'False')."
```